### PR TITLE
feat: refactor VRP to stateless

### PR DESCRIPTION
**Pull request by rsynek**
Introduces the refactored Vehicle Routing with capacity.

### CODE REVIEW

File: use-cases/vehicle-routing-capacity/src/main/java/org/acme/vehiclerouting/domain/VehicleRoutePlan.java [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+
+    @PlanningScore
+    private HardSoftLongScore score;
+
+    private Location southWestCorner;
+    private Location northEastCorner;
+
+    private SolverStatus solverStatus;
+
+    private String scoreExplanation;
```
Comment (ge0ffrey) [inline use-cases/vehicle-routing-capacity/src/main/java/org/acme/vehiclerouting/domain/VehicleRoutePlan.java:49]: Interesting and good! This is a step forward, so let's add this indeed, but I do think we need a broader discussion to figure out the best way to do this.

File: use-cases/vehicle-routing-capacity/src/main/java/org/acme/vehiclerouting/domain/solver/DepotAngleCustomerDifficultyWeightFactory.java [hunk 2]
```diff
@@ -5,21 +5,22 @@
...
 public class DepotAngleCustomerDifficultyWeightFactory
-        implements SelectionSorterWeightFactory<VehicleRoutingSolution, Customer> {
+        implements SelectionSorterWeightFactory<VehicleRoutePlan, Customer> {
 
     @Override
-    public DepotAngleCustomerDifficultyWeight createSorterWeight(VehicleRoutingSolution vehicleRoutingSolution,
-            Customer customer) {
-        Depot depot = vehicleRoutingSolution.getDepotList().get(0);
+    public DepotAngleCustomerDifficultyWeight createSorterWeight(VehicleRoutePlan vehicleRoutePlan,
+                                                                 Customer customer) {
```
Comment (ge0ffrey) [inline use-cases/vehicle-routing-capacity/src/main/java/org/acme/vehiclerouting/domain/solver/DepotAngleCustomerDifficultyWeightFactory.java]: (very soft) Weird code style?

File: use-cases/vehicle-routing-capacity/src/main/java/org/acme/vehiclerouting/rest/VehicleRoutingSolverException.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
+package org.acme.vehiclerouting.rest;
+
+import jakarta.ws.rs.core.Response;
+
+public class VehicleRoutingSolverException extends RuntimeException {
```
Comment (ge0ffrey) [inline use-cases/vehicle-routing-capacity/src/main/java/org/acme/vehiclerouting/rest/VehicleRoutingSolverException.java]: I am ok with adding it for now, but I believe that custom Exceptions and ExeptionMapper etc are overkill for quickstarts. Less code is more easier to grok for users that quick start.

**Review by ge0ffrey — APPROVED** [review]

File: use-cases/vehicle-routing-capacity/src/main/java/org/acme/vehiclerouting/domain/VehicleRoutePlan.java (hunk 1 above)
Comment (rsynek) [reply use-cases/vehicle-routing-capacity/src/main/java/org/acme/vehiclerouting/domain/VehicleRoutePlan.java:49]: This was already in place; but I agree we need to return the data structured; the justification String is just a poor man's solution.
We might have a sub-resource: `/route-plans/UUID/explanation` or even something richer.

File: use-cases/vehicle-routing-capacity/src/main/java/org/acme/vehiclerouting/rest/VehicleRoutingSolverException.java (hunk 3 above)
Comment (rsynek) [reply use-cases/vehicle-routing-capacity/src/main/java/org/acme/vehiclerouting/rest/VehicleRoutingSolverException.java]: I see your point, but the `cUrl` experience is just so much better. If everybody thinks it's an overkill, I will remove it, though.

**Pull request merged**